### ISSUE TITLE
Infra support to run a "mock" release of the engine with golden file verification

### ISSUE THREAD

**Issue by matanlurey** [flutter/flutter#146078]
### Type of Request

infra task

### Infrastructure Environment

LUCI + Cocoon? Not 100% sure.

### What is happening?

In https://github.com/flutter/flutter/issues/144835, @gaaclarke and I plan to run a "mock" release of `flutter/engine`, for the first time with golden-file tests enabled, and test out our [proposed workflow](https://github.com/flutter/engine/tree/main/testing/skia_gold_client#release-testing).

### Steps to reproduce

N/A

### Expected results

I think I need infra support:

1. What process should I follow to create a "mock" release branch?
2. How can I enable golden-file tests/`GOLDCTL` for this branch (it's disabled today IIRC)

/cc @zanderso

**Comment by christopherfujino** [comment]
> > What process should I follow to create a "mock" release branch?
> 
> One thing we can try is pick an `old` enough branch (which we are sure that we won't use), and run the workflow there. /cc @XilaiZhang @Jasguerrero who I believe just recently tried something similar for a recipe branch.

https://github.com/flutter/engine/tree/flutter-3.18-candidate.18 is the branch Xilai and I were using, you can use it also. There will not be a future release from that branch.

**Comment by christopherfujino** [comment]
If the code you need is on that branch...

**Comment by christopherfujino** [comment]
> It is not (very recent PRs).

@matanlurey if you share the earliest commit that will qualify, I can find you an unused branch.

**Comment by matanlurey** [comment]
> > It is not (very recent PRs).
> 
> @matanlurey if you share the earliest commit that will qualify, I can find you an unused branch.

https://github.com/flutter/engine/pull/51739 (`8630c4b`).

**Comment by christopherfujino** [comment]
> > > It is not (very recent PRs).
> > 
> > 
> > @matanlurey if you share the earliest commit that will qualify, I can find you an unused branch.
> 
> [flutter/engine#51739](https://github.com/flutter/engine/pull/51739) (`8630c4b`).

Ahh ok, that's too fresh for Keyong's approach (that's newer than any of our existing beta releases, so we can't confirm any of the roll branches won't be published). You could open a PR and run presubmits, but I'm guessing you want to test postsubmit too?

**Comment by matanlurey** [comment]
Right, I need to land and then open up cherry-picks against it, etc, similar to a real release branch.

**Comment by matanlurey** [comment]
I don't mind doing those steps. It's also 4:45 and I won't do them today anyway.

/cc @gaaclarke maybe we can chat for 5m tomorrow and get your take.

**Comment by matanlurey** [comment]
Temporarily assigning to myself

**Comment by matanlurey** [comment]
@XilaiZhang and I are going to pair on this together next week.

**Comment by gaaclarke** [comment]
Notes:
- So, the steps chris gave seemed to explain how to branch the recipes repository and link it to an engine branch.  We created a branch of the engine named  [flutter-test-golden-1](https://github.com/flutter/engine/tree/flutter-test-golden-1), but there needs to be a corresponding branch in the recipes repo that **has the exact same name**.  After those branches are created, there was a build triggered apropos the `ci.yaml`.  If you post a PR to the branched engine before that builder is finished, the `ci.yaml validation` presubmit check will fail to execute.
- We need to verify with Matan what is considered a valid version number for `.engine-release.version`.  In this case I would have liked to have added latin characters to the version number, but it seems from the comment that that isn't supported?  To be safe I chose `0.0.1`.

**Comment by gaaclarke** [comment]
The process seemed to work correctly up until the skia gold bot.  Goldens were generated with the proper name addition.  The bot didn't seem to pick up the new images though: https://github.com/flutter/engine/pull/52005#issuecomment-2046149513

I'm not sure what the proper course of action at this point and need to mind meld with @matanlurey and @zanderso when they are both around.

**Comment by gaaclarke** [comment]
Looks like we need a change in cocoon.  This logic appears to be skipping the skia gold bot reporting a diff.

https://github.com/flutter/cocoon/blob/9dd49cd93309b4d737d244e34d699f448e6ef17b/app_dart/lib/src/request_handlers/push_gold_status_to_github.dart#L86

I'm not sure if manually going in and accepting those goldens would be fine.  We probably don't want to do that as part of the process though since it would be easy to forget and mess up.

cc @matanlurey

**Comment by gaaclarke** [comment]
Everything has been verified to work in https://github.com/flutter/engine/pull/52005

The skia gold bot is running on engine release branches and the produced golden tests have the release name appended to them.

**Comment by gaaclarke** [comment]
I'm going to mark this as closed.  I'm convinced we are good.

**Comment by gaaclarke** [comment]
The mock test executed correctly, but on the next release the PR that sets the release name did not seem to reap the golden results like what happened in the mock test: https://github.com/flutter/engine/pull/53148

As a result the golden tests were removed from that release in https://github.com/flutter/engine/pull/53163.

We should run the mock release again to see if the recipes have regressed or to try to identify an x-factor between the test setup and release setup.

@christopherfujino has more context for this decision here: https://github.com/flutter/engine/pull/53163#issuecomment-2145802013

**Comment by matanlurey** [comment]
@gaaclarke I am going to assume you're taking the lead on this, but let me know if you'd like to pair/chat it out.

**Comment by gaaclarke** [comment]
I looked through the code a bit and I believe we are running into an x-factor between the environment where the mock release can be run and the actual releases.  I don't think doing another mock release will help.  I've filed a new bug that outlines what I know of the issue: https://github.com/flutter/flutter/issues/149670
